dias/analyzers/sensitivity_analyzer.py

```diff
@@ -0,0 +1,273 @@
+
+                t0       = time.time()
+                bflag    = (bdata.weight[:] > 0.0).astype(np.float32)
+                bvar     = tools.invert_no_zero(bdata.weight[:])
+                
+                # Loop over polarizations
+                for ii, pol in enumerate(polstr):
+
+                    self.logger.info("Processing Pol %s" % pol)
+
+                    pvar   = bvar[:, prod[pol], :]
+                    pflag  = bflag[:, prod[pol], :]
+                    pcnt   = cnt[np.newaxis, prod[pol], :]
+                    pscale = scale[pol][np.newaxis, :, np.newaxis]
+                
+                    var[freq_sel, ii, :]     += np.sum((pscale * pcnt)**2 * pflag * pvar, axis=1)
+                    counter[freq_sel, ii, :] += np.sum(pscale * pcnt * pflag, axis=1)
+
+                self.logger.info("Took %0.1f seconds to process this block." % (time.time() - t0))
+                
+                del bdata
+                gc.collect()
+            
+            # Normalize
+            inv_counter = tools.invert_no_zero(counter)
+            var        *= inv_counter**2
+
+            # Write to file
+            output_file = os.path.join(self.write_dir, "%d_%s.h5" % 
+                                      (timestamp0, self.output_suffix))
+            self.logger.info("Writing output file...")
+            
+            with h5py.File(output_file, 'w') as handler:
+        
+                index_map = handler.create_group('index_map')
+                index_map.create_dataset('freq', data=data.index_map['freq'][:])
+                index_map.create_dataset('pol', data=np.string_(polstr))
+                index_map.create_dataset('time', data=data.time)
+        
+                dset = handler.create_dataset('rms', data=np.sqrt(var))
+                dset.attrs['axis'] = np.array(['freq', 'pol', 'time'], dtype='S')
+                
+                dset = handler.create_dataset('count', data=counter.astype(np.int))
+                dset.attrs['axis'] = np.array(['freq', 'pol', 'time'], dtype='S')
+                
+                handler.attrs['instrument_name'] =self.correlator
+                handler.attrs['collection_server'] = subprocess.check_output(["hostname"]).strip()
+                handler.attrs['system_user'] = subprocess.check_output(["id", "-u", "-n"]).strip()
+                handler.attrs['git_version_tag'] = subprocess.check_output(["git", "-C", 
+                                                                            os.path.dirname(__file__), 
+                                                                            "describe", "--always"]).strip()
+
+    def get_baselines(self, indexmap, inputmap):
+        
+        prod = defaultdict(list)
+        prodmap = defaultdict(list)
+        dist = defaultdict(list)
+        conj = defaultdict(list)
+        cyl = defaultdict(list)
+        scale = defaultdict(list)
+
+        feedpos = tools.get_feed_positions(inputmap)
```

Comment:
This can be done outside of the loop over files.

Reply:
I am not sure. Do you mean calling get_baselines?

Reply:
What I thought of was moving this function call outside of the loop over files in `run()` for performance. Calling this for every file inside `get_baselines` is redundant.